src/argprocessing.cpp

```diff
@@ -262,7 +262,9 @@ process_arg(Context& ctx,
     if (argpath[-1] == '-') {
       ++argpath;
     }
-    auto file_args = Args::from_gcc_atfile(argpath);
+    auto file_args = ctx.config.compiler_type() == CompilerType::cl
```

Comment:
As far as I know, the responsfile feature in gcc was taken over one to one from msvc (I can't find the link to the mailing list entry at the moment).
At this point you should simply rename the function to e.g. 'from_responsfile' or 'from_atfile' and call the same function in both cases. 
Particularly under Windows paths with blanks can occur, so you need the logic with the quoted arguments here.